ledger-api/grpc-definitions/com/daml/ledger/api/v1/admin/party_management_service.proto

```diff
@@ -136,3 +143,31 @@ message PartyDetails {
   // Required
   bool is_local = 3;
 }
+
+message AddPartyPropertiesRequest {
+  // Required
+  string party = 1;
+  // The properties to add.
+  // Ignores adding duplicate properties.
+  // If you need to update a property, you need to delete it first using a separate call.
```

Comment:
Why have this restriction? "Upsert" semantics would be a bit easier to work with…

Reply:
I certainly would favor upsert semantics as well. How about
```
message UpdatePartyPropertiesRequest {
  string party = 1;
  repeated Property updatedProperties = 2;
  repeated Property removedProperties = 3;
}
```

Reply:
FYI
Changed the design to update properties with 'replace-all' semantics using the UpdateUser/Party call, so for the moment this discussion is not applicable.